Cabal/src/Distribution/PackageDescription/Check/Prim.hs

```diff
@@ -0,0 +1,1164 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Distribution.PackageDescription.Check.Monad
```

Comment:
```suggestion
-- Module      :  Distribution.PackageDescription.Check.Types
```

I'm pretty sure, this module should be called `...Types`. The non-type stuff in it is pretty standard kind of stuff to have in `*.Types` modules, in my experience.